cloudsim-plus/src/main/java/org/cloudbus/cloudsim/datacenters/DatacenterSimple.java

```diff
@@ -231,6 +232,9 @@ private boolean processHostEvents(final SimEvent evt) {
         } else if (evt.getTag() == CloudSimTags.HOST_REMOVE) {
             processHostRemovalRequest(evt);
             return true;
+        } else if (evt.getTag() == CloudSimTags.HOST_POWER_ON_OFF) {
+        	Host host = (HostSimple)evt.getData();
+        	host.processHostPowerOnOff();
```

Comment:
You aren't passing a boolean attribute to the method, just considering the Host active attribute will work as a flag. Every time the method is called, it's inverted. That would work perfectly, but current VmAllocationPolicies call Host.setActive(true), even if the Host is already active, because in that case, there is no side-effect.
This way, we have to pass that parameter for this new method.

Reply:
I think there is not need of parameter for host.processHostPowerOnOff() method. Inversion will work I guess.

Reply:
Yeah, the inversion will work as expected. The issue is that VmAllocationPolicy implementations don't expect this inversion and that will cause misbehavior for it. Those classes may call setActive to power a Host on. If the Host is already active, it doesn't expect it to be powered off in such a case. That is way we need that parameter.